### PR TITLE
NAS-111222 / 12.0 / Don't assume REMOVED status will be present

### PR DESCRIPTION
**Pull request by dszidi**
To test/reproduce:

- Create a pool and add a vdev of type SPARE
- physically remove one of the non SPARE disks
- Make sure Enclosure UI still displays the pools once resilver has started

### CODE REVIEW

File: src/app/core/classes/system-profiler.ts [hunk 1]
```diff
@@ -182,9 +182,7 @@ export class SystemProfiler {
         v.disks[name] = -1; // no children so we use this as placeholder
       } else if (vdev.children.length > 0) {
         vdev.children.forEach((disk, dIndex) => {
-          if (!disk.device && disk.status == 'REMOVED') {
-
-          } else {
+          if (disk.device && disk.status != 'REMOVED') {
```
Comment (RehanY147) [inline src/app/core/classes/system-profiler.ts:185]: The new condition would be true for a modal
`disk.device = true;
disk.status = 'REMOVED'`

The condition isn't exactly matched. With these values, the code would run before in the else block but won't now

**Review by RehanY147 — CHANGES_REQUESTED** [review]

**Review by RehanY147 — APPROVED** [review]
LGTM

**Comment by dszidi** [comment]
@undsoft I posted a video into the ticket comments to demo the problem as well as the fix

https://jira.ixsystems.com/browse/NAS-111222?focusedCommentId=134655&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-134655

**Comment by darkfiberiru** [comment]
image
Fix looks good.

**Review by undsoft — APPROVED** [review]
Approving based on Damian's video.

**Review by darkfiberiru — COMMENTED** [review]
Lgtm tested and shared screenshot of fixed system after drive pull

**Pull request merged**